test_python/layers/test_external_cuda_injection.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##############################################################################
+
+import tensor_comprehensions as tc
+
+import torch
+import torch.cuda
+import unittest
+
+
+class TestManualCudaInjection(unittest.TestCase):
+    def test_simple_cuda_injection(self):
+        lang = """
+        def add(float(N) A, float(N) B) -> (output) {
+            output(i) = A(i) + B(i) + 1
```

Comment:
I have the same comment as above, adding an extra ```1``` is surprising  and upon further inspection ```my_add``` below does not do it.
So you essentially have 2 implementations that don't do the same thing in the same test.
You inject the code in an unchecked way.

This test only passes because the output is not checked for correctness which is another thing I would like to see fixed.

Thanks!

Reply:
@nicolasvasilache adding such checks for all the tests is not trivial, but there are checks in place, please look at `test_tc.py`

If you want to make contribution, please send a PR.